core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -162,6 +162,35 @@ package object experimental {
     */
   trait NoChiselNamePrefix
 
+  /** Treat instances of this type as a named component, and rename using the Chisel compiler
+    * plugin.
+    *
+    * The compiler plugin by default automatically prefixes and names instances of `Data` and
+    * `Mem` recursively. This trait allows the plugin to treat arbitrary types as named
+    * components and prefix them (and objects constructed within them) when they otherwise
+    * would not be.
+    *
+    * @example {{{
+    * import chisel3._
+    * import chisel3.experimental.AffectsChiselPrefix
+    *
+    * class MyModule extends Module {
+    *   // Note: This contains a Data but is not a named component itself
+    *   class NotAData extends AffectsChiselPrefix {
+    *     val value = Wire(Bool())
+    *   }
+    *
+    *   // Name of nonData.value will be "nonData_value"
+    *   // Without AffectsChiselPrefix, the name will be "value"
+    *   val nonData = new NotAData
+    *
+    *   // Name of nonData.value will be "nonData2_value"
+    *   // Without AffectsChiselPrefix, the name will be "value_1"
```

Comment:
```suggestion
    *   // Name with AffectsChiselPrefix:    "nonData_value"
    *   // Name without AffectsChiselPrefix: "value"
    *   val nonData = new NotAData
    *
    *   // Name with AffectsChiselPrefix:    "nonData2_value"
    *   // Name without AffectsChiselPrefix: "value_1"
```
This is a great example, just a slight tweak to make the different names lineup